src/agent/coverage/src/cobertura.rs

```diff
@@ -0,0 +1,245 @@
+                .attr("branches-valid", "0")
+                .attr("branches-covered", "0")
+                .attr("branch-rate", "0")
+                .attr("complexity", "0"),
+        )?;
+        _emitter_test.write(XmlEvent::start_element("classes"))?;
+
+        _emitter_test.write(
+            XmlEvent::start_element("class")
+                .attr("name", "0")
+                .attr("filename", "C:/Users/file1.txt")
+                .attr("lines-valid","0")
+                .attr("lines-covered","0")
+                .attr("line-rate","0")
+                .attr("branches-valid", "0")
+                .attr("branches-covered", "0")
+                .attr("branch-rate", "0")
+                .attr("complexity", "0"),
+        )?;
+        _emitter_test.write(XmlEvent::start_element("lines"))?;
+
+        _emitter_test.write(
+            XmlEvent::start_element("line")
+                .attr("number", "5")
+                .attr("hits", "3")
+                .attr("branch", "false"),
+        )?;
+        _emitter_test.write(XmlEvent::end_element())?; // line
+
+        _emitter_test.write(
+            XmlEvent::start_element("line")
+                .attr("number", "10")
+                .attr("hits", "0")
+                .attr("branch", "false"),
+        )?;
+        _emitter_test.write(XmlEvent::end_element())?; // line
+
+        _emitter_test.write(XmlEvent::end_element())?; // lines
+
+        _emitter_test.write(XmlEvent::end_element())?; // class
+
+        _emitter_test.write(
+            XmlEvent::start_element("class")
+                .attr("name", "0")
+                .attr("filename", "C:/Users/file2.txt")
+                .attr("lines-valid","0")
+                .attr("lines-covered","0")
+                .attr("line-rate","0")
+                .attr("branches-valid", "0")
+                .attr("branches-covered", "0")
+                .attr("branch-rate", "0")
+                .attr("complexity", "0"),
+        )?;
+
+        _emitter_test.write(XmlEvent::start_element("lines"))?;
+
+        _emitter_test.write(
+            XmlEvent::start_element("line")
+                .attr("number", "0")
+                .attr("hits", "0")
+                .attr("branch", "false"),
+        )?;
+        _emitter_test.write(XmlEvent::end_element())?; // line
+
+        _emitter_test.write(XmlEvent::end_element())?; // lines
+        _emitter_test.write(XmlEvent::end_element())?; // class
+        _emitter_test.write(XmlEvent::end_element())?; // classes
+        _emitter_test.write(XmlEvent::end_element())?; // package
+        _emitter_test.write(XmlEvent::end_element())?; // packages
+        _emitter_test.write(XmlEvent::end_element())?; // coverage
+
+        // let check = match source_coverage_result {
+        //     Ok(source_coverage_result) => source_coverage_result,
+        //     Err(err) => return Err(err),
+        // };
+        assert_eq!(source_coverage_result.unwrap(), String::from_utf8(backing_test)?);
```

Comment:
Unsure if using .unwrap() is bad code. Checking an error some other way (similar to what I have commented out above may be better).

Reply:
Based on [this schema](https://github.com/danielpalme/ReportGenerator), I think zero (or more) elements are valid for these fields we don't have real data for.

What matters most, though, is how [ReportGenerator](https://github.com/danielpalme/ReportGenerator) renders the XML as HTML. This is because it is what [powers the ADO task](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/PublishCodeCoverageResultsV2/ThirdPartyNotices.txt#L7).

Let's investigate what happens in ADO.